src/save_index.cpp

```diff
@@ -46,6 +46,16 @@ void replace_space2underbar(std::string &name) {
 	std::replace(name.begin(),name.end(),' ','_');
 }
 
+/** Replace the spaces in the filter */
+std::string replace_filter(const std::string* filter)
+{
+    std::string filter_value = *filter;
+    std::string filter_replaced;
+    std::copy(filter_value.begin(), filter_value.end(), filter_replaced.begin());
+    replace_space2underbar(filter_replaced);
```

Comment:
This looks like a recipe for crash - you're copying a string of some length _n_ into a string of length 0, which means you'll overrun the buffer.

Try using the range constructor for strings - eg `std::string filter_replaced(filter.begin(), filter.end());`. Then you don't even need `std::copy`.